src/core/agent/agent.ts

```diff
@@ -172,68 +174,122 @@ class Agent {
-        this.getStorageService<CredentialMetadataRecord>(this.storageSession)
-      );
-      this.connectionStorage = new ConnectionStorage(
-        this.getStorageService<ConnectionRecord>(this.storageSession)
-      );
-      this.connectionNoteStorage = new ConnectionNoteStorage(
-        this.getStorageService<ConnectionNoteRecord>(this.storageSession)
-      );
-      this.notificationStorage = new NotificationStorage(
-        this.getStorageService<NotificationRecord>(this.storageSession)
-      );
-
+      const exitsUrls = await this.getAgentUrl();
+      if (
+        (!exitsUrls.url || !exitsUrls.bootUrl) &&
+        (!agentUrls.url || !agentUrls.bootUrl)
+      ) {
+        //  @TODO
+        throw new Error("Require KERI URL and Boot URL to start the agent.");
+      }
       await signifyReady();
       const bran = await this.getBran();
-      // @TODO - foconnor: Review of Tier level.
-      this.signifyClient = new SignifyClient(
-        ConfigurationService.env.keri.keria.url,
-        bran,
-        Tier.low,
-        ConfigurationService.env.keri.keria.bootUrl
-      );
 
-      this.agentServicesProps = {
-        signifyClient: this.signifyClient,
-        eventService: new EventService(),
-      };
+      if (agentUrls.url && !exitsUrls.url) {
+        // @TODO - foconnor: Review of Tier level.
+        this.signifyClient = new SignifyClient(
+          agentUrls.url,
+          bran,
+          Tier.low,
+          agentUrls.bootUrl
+        );
+        await this.signifyClient.boot();
+        await this.signifyClient.connect();
+        // Save databases
+        await this.saveAgentUrls({
+          url: agentUrls.url,
+          bootUrl: agentUrls.bootUrl as string,
+        });
+      } else {
+        this.signifyClient = new SignifyClient(
+          exitsUrls.url as string,
+          bran,
+          Tier.low,
+          exitsUrls.bootUrl
+        );
+        await this.signifyClient.connect();
+      }
 
-      this.peerConnectionStorage = new PeerConnectionStorage(
-        this.getStorageService<PeerConnectionMetadataRecord>(
-          this.storageSession
-        )
-      );
+      Agent.isOnline = true;
+
+      this.agentServicesProps.signifyClient = this.signifyClient;
 
       this.agentServicesProps.eventService.emit<KeriaStatusChangedEvent>({
         type: KeriaStatusEventTypes.KeriaStatusChanged,
         payload: {
           isOnline: Agent.isOnline,
         },
       });
+    }
+  }
 
-      try {
-        await this.signifyClient.connect();
-        Agent.isOnline = true;
-      } catch (err) {
-        await this.signifyClient.boot();
-        await this.signifyClient.connect();
-        Agent.isOnline = true;
-      }
+  async getAgentUrl(): Promise<Partial<AgentUrls>> {
+    const keriUrlRecord = await this.basicStorageService.findById(
+      MiscRecordId.KERI_URL
+    );
+    const keriBootUrlRecord = await this.basicStorageService.findById(
+      MiscRecordId.KERI_BOOT_URL
```

Comment:
The function to boot the agent can accept the boot URL as a parameter instead of reading from disk. We can still store it in the DB I think during onboarding but that would be another PR from the UI

When users are recovering an existing wallet, they will just enter the connect URL and skip the UI call to call boot (since it will already be booted).